cvat/apps/dataset_manager/bindings.py

```diff
@@ -1885,17 +1885,21 @@ def load_dataset_data(project_annotation, dataset: dm.Dataset, project_data):
             'data_root': dataset.data_path + osp.sep,
         }
 
+        data_root = None
         for dataset_item in subset_dataset:
             if dataset_item.image and dataset_item.image.has_data:
                 dataset_files['media'].append(dataset_item.image.path)
+                if data_root is None:
+                    data_root = osp.splitext(dataset_item.image.path)[0].replace(dataset_item.id, '')
             elif dataset_item.point_cloud:
                 dataset_files['media'].append(dataset_item.point_cloud)
+                if data_root is None:
+                    data_root = osp.splitext(dataset_item.point_cloud)[0].replace(dataset_item.id, '')
             if isinstance(dataset_item.related_images, list):
                 dataset_files['media'] += \
                     list(map(lambda ri: ri.path, dataset_item.related_images))
 
-        shortes_path = min(dataset_files['media'], key=lambda x: len(Path(x).parts), default=None)
-        if shortes_path is not None:
-            dataset_files['data_root'] = str(Path(shortes_path).parent.absolute()) + osp.sep
+        if len(dataset_files['media']):
+            dataset_files['data_root'] = data_root
```

Comment:
This solution doesn't seem correct. It just takes the first path, and it may not be the common prefix. Please check if [this function](https://docs.python.org/3/library/os.path.html#os.path.commonpath) can be useful here. Also, with this solution I'm still able to reproduce the error in a dataset with different directory levels.

Reply:
Fixed it. But nested folders are not possible for ImageNet at the moment.